src/Tar.jl

```diff
@@ -461,4 +481,24 @@ check_tree_hash_tarball(tarball::AbstractString) =
 
 check_tree_hash_tarball(tarball::ArgRead) = nothing
 
+function check_windows_path(
+    path  :: AbstractString,
+    parts :: AbstractVector{<:AbstractString} = split(path, r"/+"),
+)
+    for part in parts
+        isempty(part) && continue
+        ch = part[end]
+        if ch == ' ' || firstindex(part) < lastindex(part) && ch == '.'
+            error("illegal Windows char: $(repr(ch)) in $(repr(path))")
```

Comment:
These are legal file names, though doing this may make the file invisible to many virus scanner tools, and windows explorer cannot handle the filename

Reply:
Ok, I think we mostly don't care about that so we should just handle these.